RELEASE/scripts/autoscend/iotms/mr2022.ash

```diff
@@ -225,3 +225,21 @@ void juneCleaverChoiceHandler(int choice)
 			abort("unhandled choice in juneCleaverChoiceHandler");
 	}
 }
+
+void sweatpantsPreAdventure() {
+	if (!possessEquipment($item[designer sweatpants]) || !can_equip($item[designer sweatpants])) {
+		return;
+	}
+
+	int sweat = get_property("sweat").to_int();
+	int liverCleaned = get_property("_sweatOutSomeBoozeUsed").to_int();
+
+	if (sweat >= 25 && liverCleaned < 3 && my_inebriety() > 0) {
+		use_skill($skill[Sweat Out Some Booze]);
+	}
+
+	// This is just opportunistic use of sweat. This skill should be used in auto_restore.ash.
+	if (sweat >= 95 && my_mp() < my_maxmp()) {
```

Comment:
The spot to put it would be after soulsauce usage which starts on line 1747 of auto_restore.ash. At first glance, I'd say you can use the soulsauce code as inspiration for sweat MP restore. Both use a lower value resource to generate MP. I'm totally fine with this being a future PR or in this one, up to you.

Reply:
thanks!